File: backend/src/main/java/uni/bugtracker/backend/model/Session.java
package uni.bugtracker.backend.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.time.Instant;
import java.util.List;

@Data
@Entity
@Table(name = "session")
public class Session {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "project_id", nullable = false)
    private Project project;

    @NotNull
    private Boolean isActive;

    @NotNull
    private Instant startTime;

    private Instant endTime;
    private String browser;
    private String browserVersion;
    private String os;
    private String deviceType;
    private String screenResolution;
    private String viewportSize;
    private String language;
    private String userAgent;
    private String ipAddress;
    private String cookiesHash; // Hashed for privacy

    @ElementCollection
    @CollectionTable(
            name = "session_plugins",
            joinColumns = @JoinColumn(name = "session_id")
    )
    @Column(name = "plugin")
    private List<String> plugins;
}

File: frontend/widget/src/types/session.ts
export interface SessionData {
    id: number;
    startTime: string;
    lastActivity: string;
    active: boolean;

    // SessionContext fields
    browser: string;
    browserVersion: string;
    os: string;
    plugins: string[];
    screenResolution: string;
    viewportSize: string;
    cookiesHash?: string;
    timezone: string;
    language: string;
    userAgent: string;
    ipAddress?: string;

    // Additional fields
    deviceType: 'desktop' | 'mobile' | 'tablet';
    connectionType?: string;
    memoryUsage?: string;
    url: string;
    referrer?: string;
}